packages/api/cms-api/src/warnings/warning-event-subscriber.ts

```diff
@@ -53,12 +59,39 @@ export class WarningEventSubscriber implements EventSubscriber {
                 });
                 for (const node of flatBlocks.depthFirst()) {
                     const warnings = node.block.warnings();
-                    await this.warningService.updateWarningsForBlock(warnings, {
-                        rootEntityName: entity.name,
-                        rootColumnName: key,
-                        targetId: args.entity.id,
-                        rootPrimaryKey: args.meta.primaryKeys[0],
-                        jsonPath: node.pathToString(),
+
+                    await this.warningService.saveWarningsAndDeleteOutdated({
+                        warnings,
+                        type: "Block",
+                        sourceInfo: {
+                            rootEntityName: entity.name,
+                            rootColumnName: key,
+                            targetId: args.entity.id,
+                            rootPrimaryKey: args.meta.primaryKeys[0],
+                            jsonPath: node.pathToString(),
+                        },
+                    });
+                }
+            }
+
+            const emitWarnings = this.reflector.getAllAndOverride<EmitWarningsMeta>("emitWarnings", [entity]);
+            if (emitWarnings) {
+                const emitWarnings = this.reflector.getAllAndOverride<EmitWarningsMeta>("emitWarnings", [entity]);
+                const repository = this.entityManager.getRepository(entity);
+
+                const rows = await repository.find();
+
+                for (const row of rows) {
+                    const service = this.moduleRef.get(emitWarnings, { strict: false });
+
+                    await this.warningService.saveWarningsAndDeleteOutdated({
```

Comment:
shouldn't you delete outdated only once? (now you do it once for blocks and once for entities)

Reply:
Currently it's necessary to call it twice. The function currently works that it only deletes outdated with the same type, sourceInfo, but maybe my function naming is not clear enough. 

<img width="932" alt="Screenshot 2025-03-25 at 17 44 43" src="https://github.com/user-attachments/assets/06824c98-1307-407e-812d-359a5f7922e3" />


Maybe:
 - refreshWarnings
 - saveWarningsAndDeleteOutdatedOfSameInfo
 - saveWarningsAndDeleteOutdatedOfSameTypeAndSourceInfo

Reply:
So you're using the source info as an ID, right? Didn't we plan to use other information for this? 🤔

Reply:
No, I just didn't understand it 😁